conans/client/source.py

```diff
@@ -105,7 +105,9 @@ def get_sources_from_exports():
                 # First of all get the exported scm sources (if auto) or clone (if fixed)
                 _run_cache_scm(conanfile, scm_sources_folder, output)
                 # so self exported files have precedence over python_requires ones
-                merge_directories(export_folder, conanfile.folders.base_source)
+                if not hasattr(conanfile, "layout"):
```

Comment:
This was a necessary change to implement the SCM flow, as the conandata.yml happens to be copied the first thing to the cache "source" folder, so trying to do a clone in the cache in that folder fails